### PR TITLE
Change --frozen-lockfile to --immutable

### PR DESCRIPTION
**Pull request by IgorNadj**
--frozen-lockfile is deprecated

https://yarnpkg.com/cli/install#options-immutable

### CODE REVIEW

File: packages/devops/scripts/deployment/buildDeployablePackages.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ PACKAGES=$(${TUPAIA_DIR}/scripts/bash/getDeployablePackages.sh)
 
 # Install external dependencies and build internal dependencies
 cd ${TUPAIA_DIR}
-yarn install --frozen-lockfile
+yarn install --immutable
```
Comment (IgorNadj) [inline packages/devops/scripts/deployment/buildDeployablePackages.sh:15]: Tested ✅ :
Screen Shot 2023-05-16 at 4 13 19 pm

File: packages/meditrak-app/appcenter-post-clone.sh [hunk 2]
```diff
@@ -21,7 +21,7 @@ nvm use
 cd ../..
 
 # install root dependencies
-SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --frozen-lockfile
+SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --immutable
```
Comment (IgorNadj) [inline packages/meditrak-app/appcenter-post-clone.sh:24]: Tested in appcenter ✅ :

File: packages/devops/ci/validation.Dockerfile [hunk 3]
```diff
@@ -15,7 +15,7 @@ WORKDIR /tupaia
 COPY . ./
 
 # run yarn without building, so we can cache node_modules without code changes invalidating this layer
-RUN SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --frozen-lockfile
+RUN SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --immutable
```
Comment (IgorNadj) [inline packages/devops/ci/validation.Dockerfile:18]: Tested in CI/CD ✅ 
See the build for this commit

File: packages/devops/ci/tupaia-ci-cd.Dockerfile [hunk 4]
```diff
@@ -101,7 +101,7 @@ RUN mkdir -p ./packages/web-frontend
 COPY packages/web-frontend/package.json ./packages/web-frontend
 
 # run yarn without building, so we can cache node_modules without code changes invalidating this layer
-RUN SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --frozen-lockfile
+RUN SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --immutable
```
Comment (IgorNadj) [inline packages/devops/ci/tupaia-ci-cd.Dockerfile:104]: Tested in CI/CD ✅ 
See the build for this commit

**Review by rohan-bes — APPROVED** [review]

**Pull request merged**